package/MDAnalysis/lib/nsgrid.pyx

```diff
@@ -901,6 +901,8 @@ cdef class FastNS(object):
                                     if probe[m] > self.box.c_pbcbox.box[m][m]:
                                         check = False
                                         break
+                                    if probe[m] == self.box.c_pbcbox.box[m][m]:
```

Comment:
I am always wary of testing equality for floats. Is this a case when it is justified to do so?

Or should this also do a comparison within a certain epsilon?

Reply:
Yes @orbeckst, you are right. It is not a good idea to compare the equality of floats. While debugging, I found the error here which seemed to get resolved by this correction. But I agree, it wasn't robust enough.

Reply:
As far as the algorithm is concerned, this specific patch tries to find the neighboring cell by moving in all 3 directions (27 combinations using probe). The problem here was that probe was moving out of the box dimensions (not quite, very close to the edge of the box).